packages/twenty-server/src/modules/contact-creation-manager/services/create-company-and-contact.service.ts

```diff
@@ -155,31 +160,43 @@ export class CreateCompanyAndContactService {
       throw new Error('Object metadata not found');
     }
 
-    const domainNameFieldMetadata = await this.fieldMetadataRepository.findOne({
-      where: {
-        workspaceId: workspaceId,
-        standardId: COMPANY_STANDARD_FIELD_IDS.domainName,
-      },
-    });
+    // In some jobs the accountOwner is not populated
+    if (!connectedAccount.accountOwner) {
+      const workspaceMemberRepository =
+        await this.twentyORMGlobalManager.getRepositoryForWorkspace(
+          workspaceId,
+          WorkspaceMemberWorkspaceEntity,
+        );
+
+      const workspaceMember = await workspaceMemberRepository.findOne({
+        where: {
+          id: connectedAccount.accountOwnerId,
+        },
+      });
+
+      if (!workspaceMember) {
+        throw new Error(
+          `Workspace member with id ${connectedAccount.accountOwnerId} not found in workspace ${workspaceId}`,
+        );
+      }
 
-    const companyDomainNameColumnName =
-      domainNameFieldMetadata?.type === FieldMetadataType.LINKS
-        ? 'domainNamePrimaryLinkUrl'
-        : 'domainName';
+      connectedAccount.accountOwner = workspaceMember;
+    }
```

Comment:
We can modify the code to always pass the connected account with the account owner to the job. But it will be easier once all the raw queries are replaced.